### ISSUE TITLE
Levelling and Compacting tractor does not wait for trailer to unload

### ISSUE THREAD

**Issue by Symbob** [Courseplay/courseplay#1832]
When the levelling and compacting tractor stops at the wait point for trailers to tip, as soon as the trailer starts to reverse in to the bunker the levelling tractor starts again before the trailer has been able to tip.  This causes the tractors to collide with each other.  Sometimes the trailer will stop before it hits the leveller.  The leveller will travel a length of the bunker then attempt to move to the waiting spot again, but the trailer is now stopped in the way.  I've seen this problem mentioned by some other people on the FS-UK forum too.

**Comment by dkfiremann** [comment]
The important point here is to show pictures. 

Where is the waiting point placed compared to the bunker walls? 
Where is the tractors route compared to the bunker wall?

**Comment by pops64** [comment]
Log with debug channel 10? Try increasing your detection radius.

**Comment by Symbob** [comment]
I tried to upload a video but the only software I've got makes crazy large files, too large to be uploaded.  Here's a few pictures which I hope will demonstrate the problem.  In picture 01 the trailer is waiting for the leveller to stop at the wait point and the leveller has just reached the wait point and waits for the trailer.  In pic 03 the leveller has already left the wait point before the trailer has even started the reversing manoeuvre.  The trailer and leveller then reverse towards each other (following their courses) until the trailer stops again just before hitting the leveller.  Sometimes on other recorded courses they will actually crash in to each other.  In this example the leveller then continues to level the bunker for one pass then attempts to go back to it's waiting point, but the trailer is now in the way and prevents it from doing so.  You can see the levellers route in the pictures, and you can see the trailers route in pic 11.

EDIT
Sorry.  I've tried uploading these pictures several times but it just keeps telling me that "Something went really wrong, and we can't process that file".  I've tried uploading just one at a time but it won't accept it.  Is there a particular format the pics need to be in?  These are just standard jpg's.

**Comment by mike3112** [comment]
This happens to me too but also the leveller will tell the trailer to stop till it reaches the wait out the way but then it doesn't tell the trailer to set off again. I have no log but seemed a good place to bring it up

**Comment by dkfiremann** [comment]
@Symbob how big is the pictures in kb?
Have you tried with another browser?

Remember to show the complete route of the leveller in the picture.

**Comment by Symbob** [comment]
I'm not going to be much help to you I'm afraid as I don't seem to be able to upload any files at all, not pictures or my log.  

Increasing the detection radius, as pops64 suggested, did have improve things slightly.  I changed the detection radius from 50 to 100m.  The leveller moves to it's waiting point, the trailer starts to move towards the bunkers, but again the leveller leaves it's waiting point and starts levelling again.  It completes one pass of the bunker then moves back to it's waiting point.  The trailer has stopped again to wait for the leveller to get back to its wait point, but because of the increased detection radius it has stopped much further away.  Once the leveller reaches its wait point the trailer proceeds to the bunker and tips it's load while the leveller waits for him to finish.  So at least the trailers were able to tip, but that doesn't explain why the leveller always decides to leave it's wait point to carry out one last pass of the bunker once the trailers have been signalled to come in and tip.

**Comment by Symbob** [comment]
The pictures are just 190k each.  I'll switch to another browser and try again.

**Comment by Symbob** [comment]
[log.txt](https://github.com/Courseplay/courseplay/files/809072/log.txt)

**Comment by Symbob** [comment]
Ah, we're getting somewhere now!!  See log above.

**Comment by Symbob** [comment]
Here's the pictures

![11](https://cloud.githubusercontent.com/assets/18528850/23436164/6d58b3a8-fe01-11e6-9d19-1b29369fcbfe.jpg)
![01](https://cloud.githubusercontent.com/assets/18528850/23436115/4d609d72-fe01-11e6-8ad9-62f86c727bf5.jpg)
![02](https://cloud.githubusercontent.com/assets/18528850/23436116/4d680c2e-fe01-11e6-976e-6ff33d65f883.jpg)
![03](https://cloud.githubusercontent.com/assets/18528850/23436117/4d6bdb42-fe01-11e6-909a-c87ffbaacd2b.jpg)
![04](https://cloud.githubusercontent.com/assets/18528850/23436118/4d73b3a8-fe01-11e6-992c-be9b8741f343.jpg)
![05](https://cloud.githubusercontent.com/assets/18528850/23436119/4d79057e-fe01-11e6-95d2-53ac8d52940f.jpg)
![06](https://cloud.githubusercontent.com/assets/18528850/23436120/4d7f5474-fe01-11e6-8332-2bde4a3d926e.jpg)
![07](https://cloud.githubusercontent.com/assets/18528850/23436121/4d8a7cbe-fe01-11e6-8b3e-cb641afebc0f.jpg)
![08](https://cloud.githubusercontent.com/assets/18528850/23436122/4d8bfc06-fe01-11e6-8d9e-4896d8243d65.jpg)
![09](https://cloud.githubusercontent.com/assets/18528850/23436123/4d9427aa-fe01-11e6-84f6-17e8c3e25ade.jpg)
![10](https://cloud.githubusercontent.com/assets/18528850/23436124/4d9a5e5e-fe01-11e6-88df-ca6b6bc5c9e7.jpg)

**Comment by pops64** [comment]
Try again setting up mode 10 as follows. https://github.com/Courseplay/courseplay/files/679279/setup_mode10_course.pdf Start point should be towards the edge of the silo and 2nd should just out side of it. No all the way in the back.

**Comment by Symbob** [comment]
OK, this is starting to look like user error.  I recorded a course as advised and everything appears to be working OK.  I recorded a few courses in fact, making sure the start point was just in the bunker and the second just outside and everything worked fine except for one time when the leveller did the same thing again, moved away from its wait point and started working again.  I rerecorded the course and then it behaved as it should, so this looks to me as though its probably my fault for not recording the course properly.  

I'll keep monitoring the problem and see if it happens again.

Thanks for the help and thanks for an awesome mod.
